src/libraries/System.Linq/src/System/Linq/Range.SpeedOpt.cs

```diff
@@ -16,23 +19,55 @@ public override IEnumerable<TResult> Select<TResult>(Func<int, TResult> selector
 
             public int[] ToArray()
             {
-                int[] array = new int[_end - _start];
-                Fill(array, _start);
+                int start = _start;
+                int[] array = new int[_end - start];
+                Fill(array, start);
```

Comment:
Does reading _start once instead of twice make a meaningful difference?  Same question below for _end and _start.

Reply:
Do you have evidence that it actually helps?  From looking at the codegen, this actually looks worse.  Please revert.